### DISCUSSION TITLE
Buat project planning

### DISCUSSION THREAD

**Discussion by jufianto** [KongkowITPekanbaru/masjid-network#4] [Ideas]
harus dibuat markdown file / page khusus untuk fitur apa aja yang akan ada di mesjid-network ini. 

beberapa fitur yang masih terpikirkan 

* pendataan aset mesjid 
* pencatatan kas mesjid 
* jadwal kajian 

beberapa micro yang terpikir 
* micro users (manajemen user login)
* micro jadwal kajian 
* micro pencatatan aset
* micro pencatatan kas

## Todoist 

* buat user stories (usecase)
* flowchart setiap proses

**Comment by ahmad-reza619** [comment]
Mungkin boleh tambahan ya kayak fitur jadwal solat mungkin 🤔

**Reply by jufianto** [reply]
kalau gitu cari api yang jadwal sholat setiap kota mungkin, btw nanti tujuannya juga biar bisa ditampilkan di layar besar gitu / monitor gitu

**Reply by Eore** [reply]
Nah ini sectionnya beda lg, productivity mngkin? Atosementara masuk misc dlu

**Reply by ibnudaqiqil** [reply]
saya rasa ga, karena ada fitur pencatatan uang. jadi bisa sebagai reporting screen

**Reply by far1023** [reply]
> ... btw nanti tujuannya juga biar bisa ditampilkan di layar besar gitu / monitor gitu

semua yang ditampilkan data server online mas?

**Reply by Eore** [reply]
> saya rasa ga, karena ada fitur pencatatan uang. jadi bisa sebagai reporting screen

oh maksudnya, section fitur nya, jdi nnti di bagi dlu fitur2 ini masuk section apa, ini ngebagi responsibility service sih, untuk nnti tampilannya gmn, dirancang dlu mockup nya,  klo skrng paling rencanain MVP nya dlu @ibnudaqiqil 

> > ... btw nanti tujuannya juga biar bisa ditampilkan di layar besar gitu / monitor gitu
> 
> semua yang ditampilkan data server online mas?

bisa jadi, skrng masi tahap perancangan MVP nya mau gmn @far1023

**Comment by ahmad-reza619** [comment]
bertanya juga bang penjelasan lebih detailnya, kalau yang pencatatan kas dan jadwal kajian sih bisa za mengerti, tapi kalau pendataan aset masjid ini gimana ya maksudnya?

**Reply by jufianto** [reply]
seperti asset kipas, ac, sapu dan lain lain.
kayak harus di list gitu aja sih. 
nanti bisa di cek secara berkala juga untuk biaya perawatan, ada semacam reminder gitu

**Reply by Eore** [reply]
Brrti nnti ad 2 section, assets sm finance

**Comment by Eore** [comment]
Brrti yg kebayang skrng servicenya
- Service Finance
- Service Assets
- Service Productivity (lain2 nya yg masuk fitur pendukung) 
- Service Integration (ini nnti nya jdi service jembatan antara dunia luar) 
- Service Organization (user nnti masuk ksini, jdi di tiap masjid kan pny user pengurus)
- Gateway

**Reply by jufianto** [reply]
gateway itu lebih kemana ya mas? belum kebayang

**Reply by Eore** [reply]
Karna servicrnya independent, jdinya FE nge consume via gateway, nnti smua service di route dsitu @jufianto

**Reply by ahmad-reza619** [reply]
hmm apa ga overkill ya bang kalau gitu rasanya?
wkwk bikin dia nanti jadi lebih kompleks ga sih itu? @Eore

**Reply by Eore** [reply]
Itu klo emang startnya mau micro sih, ya klo mau monolitik ga perlu bgituan, brrti si service itu bkn literally service, tpi abstraksi @ahmad-reza619

**Reply by ahmad-reza619** [reply]
hmm coba mikro deh, monolitik udah terlalu sering wkwk

**Reply by Eore** [reply]
MVP nya mau kas dlu kan ya? Ato aset dlu? 
Klo kas dlu kyna prlu org yg tau mslh kas masjid, aku tau bagian luar nya aja, ky hitung uang yg masuk, trus nnti liat perencanaan cash out

**Reply by jufianto** [reply]
kayaknya butuh MVP design dulu sebelum mulai kita, minimal flowchart sama design DB. 
btw seminggu ini kita tuangkan aja dulu apa yang rasanya fitur yang kita perlukan.

**Reply by Eore** [reply]
Aku vote kas dlu sih untuk MVP, rasa aku lbih membantu buat pengurus dan umat

**Reply by kurnianto12** [reply]
service integration yang kedunia luar itu seperti jadwal kajian rutin tiap masjid aja mas er .? @Eore

**Reply by Eore** [reply]
klo kajian rutin nnti bisa masuk ke fitur, yg kluar itu misal nnti ad integrasi ke dompet dhuafa misal, atau mau bayar infak mesjid nya pake QR code, jdi integrasi ke pembayaran digital @kurnianto12

**Comment by febrihidayan** [comment]
Ini multi masjid atau hanya untuk masing-masing masjid?

**Reply by jufianto** [reply]
multi

**Reply by febrihidayan** [reply]
@jufianto Apakah bisa saya membuat projek ini menggunakan Laravel? dengan repo yang berbeda

**Comment by ibnudaqiqil** [comment]
Untuk pencatatan keuangannya kayaknya bagus jika dibuat kategori2 khusus, misalnya zakat, infak, dana pembanguanan. jadi bisa ketahuan berapa uang zakat total, mana masjid yang kekurangan dll.

**Reply by kurnianto12** [reply]
iya ntar kayaknya diberi label aja @Eore

**Reply by Eore** [reply]
untuk jenis kas dan pelabelan udh ad di ide perancangan kok, tinggal cash flow nya nnti ad flow khsus ato ga blum tau

**Comment by ibnudaqiqil** [comment]
- Mungkin bisa dibuat model kitabisa.com yang sederhana.. misalnya menerima donasi untuk renovasi lantai, lengkap dengan pelaporan
- manajemen dan pendaftaran qurban. Beberapa masjid ada yang sistem cicilan sepanjang tahun, dsb.

**Reply by Eore** [reply]
bole jg, noted dlu, itu untuk pengembangannya brrti nnti

**Comment by Eore** [comment]
![erd](https://www.nomnoml.com/image.svg?source=%5Briwayat_kas%7C*id%3Auuid%3Bid_organisasi%3Auuid%3Bnama_kas%3Astring%3Blabel%3Astring%3Bjumlah%3Anumeric%3Bmetadata%3Ajson%5D%0A%5Btotal_kas%7C*id%3Auuid%3Bid_organisasi%3Auuid%3Btotal%3Ajson%5D%0A%5Borganisasi%7C*id%3Auuid%3Bnama%3Astring%3Bmetadata%3Ajson%5D%0A%5Buser%7C*id%3Auuid%3Bid_organisasi%3Auuid%3Busername%3Astring%3Bpassword%3Astring%3Bpermission%3A%5C%5B%5C%5Dstring%5D%0A%0A%5Btotal_kas%5D-%3E%5Borganisasi%5D%0A%5Buser%5D-%3E%5Borganisasi%5D%0A%5Briwayat_kas%5D-%3E%5Borganisasi%5D)

sementara yg kebayang

**Reply by Eore** [reply]
beberapa tambahan data akan masuk ke metadata (nama nya sementara itu dlu)

**Reply by aflahu** [reply]
Maaf mas @eore table total kas untuk apa ya? Soalnya saya pikir kita bisa menggunakan sum, itu lebih aman

**Reply by Eore** [reply]
Yap itu bnr kok, cm nnti bakal ad multi masjid, trus jumlah data nya bakal nambah, brrti klo hrus sum untuk ngambil data nya, bakal berat di db, jdi dibuat 1 table itu, nnti tiap ada kas masuk, dka akan update itu jg

Btw open for discussion untuk itu, ini kan ide awal aku aja, klo ad metode lain yg lbih baik, bolee

**Reply by ibnudaqiqil** [reply]
akan kurang efektif seperti itu, misalnya mau ambil data bulan agustus, awalnya berapa, akhirnya berapa
gimana kalau model akuntansi, debit, kredit, total.  
coba dibayangin dlu luarannya nanti gmn? baru di optimasi.
biasanyakan kalo mesjid itu laporannya bulanan, berapa pemasukan, berapa pengeluaran, itu biasanya ditampilkan per-group.

**Reply by ibnudaqiqil** [reply]
btw, kalau pertimbangannya performa kenapa menggunakan uuid? knp ga autocreament? ada rencana terdistribus? composite key?

**Reply by aflahu** [reply]
> akan kurang efektif seperti itu, misalnya mau ambil data bulan agustus, awalnya berapa, akhirnya berapa
> gimana kalau model akuntansi, debit, kredit, total.  
> coba dibayangin dlu luarannya nanti gmn? baru di optimasi.
> biasanyakan kalo mesjid itu laporannya bulanan, berapa pemasukan, berapa pengeluaran, itu biasanya ditampilkan per-group.
> 
> 

Mantap pemikirannya, untuk Total_kas akan dipakai untuk lihat total saldo sekarang, dan kita tetap bisa mendapatkan data pembukuan perbulan dengan select riwayat kas

**Reply by aflahu** [reply]
> btw, kalau pertimbangannya performa kenapa menggunakan uuid? knp ga autocreament? ada rencana terdistribus? composite key?

@eore ohya kenapa totalkasnya tipe JSON ya?

**Reply by Eore** [reply]
> 
> 
> akan kurang efektif seperti itu, misalnya mau ambil data bulan agustus, awalnya berapa, akhirnya berapa
> gimana kalau model akuntansi, debit, kredit, total.
> coba dibayangin dlu luarannya nanti gmn? baru di optimasi.
> biasanyakan kalo mesjid itu laporannya bulanan, berapa pemasukan, berapa pengeluaran, itu biasanya ditampilkan per-group.

yg ini bisa jg, perlu org yg pernah urus kas masjid biar bisa tau alur nya bagaimana yg tepat, klo yg kebayang sm aku sesimple catat aja uang masuk kluar yg ada, klo masuk positif, klo keluar negatif (mngkin klo di akuntansi pake debit kredit tdi), si `total_kas` itu bakal jdi tutup buku nya

btw thank for the reminder, early optimization is the root of all evil emang, kindly drop your idea design in this discussion tho



> 
> 
> btw, kalau pertimbangannya performa kenapa menggunakan uuid? knp ga autocreament? ada rencana terdistribus? composite key?

klo ini, uuid dsitu ga harus uuid ya (bisa di ganti nama field nya id, yg pnting field itu bakal jdi primary key yg unik), bisa jg auto generated string format, untuk smentara uuid dlu, karna ckup standard
yep, serial emang lbih bgus dri segi performa, cm klo untuk historical data kyna ga cocok, soal nya serial pny batasan (untuk big int memang batas nya gede, tapi tetep berbatas), kecuali mngkin, klo udh smpe brp tahun, data lama di archive trus counternya mulai dri awal lg, serial bisa jadi pilihan yang bagus
untuk terdistribusi, mngkin bisa mngkin ga, karna jatuh nya kan projek komunitas aja, bisa buat ajang bljr, poc, atau eksperimen, target nya emang hnya untuk buat open source project dri kita untuk kita, dan sebisa mngkin terdelivery, ya mngkin bisa kedepannya jdi terdistribusi, 

> 
> 
> > btw, kalau pertimbangannya performa kenapa menggunakan uuid? knp ga autocreament? ada rencana terdistribus? composite key?
> 
> @Eore ohya kenapa totalkasnya tipe JSON ya?

ini klo ga salah kebayang ky total nya macem2, jdi aku store aja in object (key nya nama total nya, value nya si total nya), but feel free to propose your idea

**Comment by ahmad-reza619** [comment]
Design untuk tampilan yang saya buat:
https://www.figma.com/file/99NItN21AeAtgqGfLQ0paj/Masjid-Network?node-id=0%3A1

masih on progress sih ini, tapi saya open buat saran & kritik 😄

**Reply by achimonchi** [reply]
Keknya, untuk design ini perlu disatuin ga jak?
soalnya kmaren bg putra udah buat designnya juga..
wkwkwk

**Reply by ahmad-reza619** [reply]
iya kayaknya, cuma figma untuk team cuma bisa 2 user kalo ga salah

**Comment by Indrafsyah** [comment]
wah aku baca seru jga,
ini aku ijin nimbrung
untuk asset management kaya nya emang di butuhkan di mesjid " yang memang asset nya sudah banyak, biar bisa rencanain kapan harus di bersikan dan kapan harus di service..

tapi yang aku perhatiin yang akan bantu banget manajement mesjid itu di cash flow sih,karena dengan adanya dashboard berapa uang masuk dan uang keluar, jadi mempermudah banget pengurus mesjid dalam membuat laporan setiap hari jumat

Selain itu kadang kan mesjid jga ada kebutuhan bisa di tampilin jga berapa biaya yang sudah terpenuhi dan berapa yang belum, jadi jamaah jga lebih aware dan harapan nya bisa antusias untuk memakmurkan mesjid

kayanya konsep yang bisa kita ATM itu di buku kas atau aplikasi semacam nya sih buat contek flownya

**Reply by Eore** [reply]
nah iya keuangan emang masuk salah satu feature nya nnti, cm kmrn masi blm dpt yg bgus untuk pencatatan keuangannya (perlu org yg tau akuntansi), jadi nya target MVP awal nya aset dlu @Indrafsyah
